sdk/identity/Azure.Identity/tests/OnBehalfOfCredentialTests.cs

```diff
@@ -166,5 +166,50 @@ public async Task SendCertificateChain([Values(true, false)] bool sendCertChain)
             var token = await client.GetTokenAsync(new TokenRequestContext(MockScopes.Default), default);
             Assert.AreEqual(token.Token, expectedToken, "Should be the expected token value");
         }
+
+        [Test]
+        public async Task ValidatesClientAssertionIsCorrect()
+        {
+            var expectedToken = Guid.NewGuid().ToString();
+            var expectedClientAssertion = Guid.NewGuid().ToString();
+            TransportConfig transportConfig = new()
+            {
+                TokenFactory = req => expectedToken,
+                RequestValidator = req =>
+                {
+                    if (req.Content != null)
+                    {
+                        var stream = new MemoryStream();
+                        req.Content.WriteTo(stream, default);
+                        var content = new BinaryData(stream.ToArray()).ToString();
+                        Assert.That(content, Does.Contain($"client_assertion={expectedClientAssertion}"));
+                    }
+                }
+            };
+            var factory = MockTokenTransportFactory(transportConfig);
+            var _transport = new MockTransport(factory);
+            var _pipeline = new HttpPipeline(_transport, new[] { new BearerTokenAuthenticationPolicy(new MockCredential(), "scope") });
+            var certificatePath = Path.Combine(TestContext.CurrentContext.TestDirectory, "Data", "cert.pfx");
+            var mockCert = new X509Certificate2(certificatePath);
+
+            options = new OnBehalfOfCredentialOptions
+            {
+                AuthorityHost = new Uri("https://localhost"),
```

Comment:
Just out of curiosity, why do we use localhost as the authority for the tests?

Reply:
no specific reason. It could be anything.